src/routes/transactions/mappers/multisig-transactions/multisig-transaction-details.mapper.ts

```diff
@@ -0,0 +1,208 @@
+    );
+    const txStatus = this.statusMapper.mapTransactionStatus(transaction, safe);
+    const detailedExecutionInfo = await this._mapMultisigExecutionDetails(
+      chainId,
+      transaction,
+      txStatus,
+      safe,
+    );
+    const recipientAddressInfo = await this._getRecipientAddressInfo(
+      chainId,
+      transaction.to,
+    );
+
+    return {
+      safeAddress: safe.address,
+      txId: `${MULTISIG_TRANSACTION_PREFIX}${TRANSACTION_ID_SEPARATOR}${safe.address}${TRANSACTION_ID_SEPARATOR}${transaction.safeTxHash}`,
+      executedAt: transaction.executionDate?.getTime() ?? null,
+      txStatus,
+      txInfo,
+      txData: new TransactionData(
+        transaction.data,
+        transaction.dataDecoded,
+        recipientAddressInfo,
+        transaction.value,
+        transaction.operation,
+        isTrustedDelegateCall,
+        isEmpty(addressInfoIndex) ? null : addressInfoIndex,
+      ),
+      txHash: transaction.transactionHash,
+      detailedExecutionInfo,
+      safeAppInfo,
+    };
+  }
+
+  private async _mapMultisigExecutionDetails(
+    chainId: string,
+    transaction: MultisigTransaction,
+    txStatus: TransactionStatus,
+    safe: Safe,
+  ): Promise<MultisigExecutionDetails> {
+    const gasToken = transaction.gasToken ?? NULL_ADDRESS;
+    const gasTokenInfo =
+      gasToken != NULL_ADDRESS
+        ? await this.tokenRepository.getToken(chainId, gasToken)
+        : null;
+    const executor = transaction.executor
+      ? await this.addressInfoHelper.getOrDefault(chainId, transaction.executor)
+      : null;
+    const signers = safe.owners.map((owner) => new AddressInfo(owner));
+    const confirmationsRequired =
+      transaction.confirmationsRequired ?? safe.threshold;
+    const confirmations = !transaction.confirmations
+      ? []
+      : transaction.confirmations.map(
+          (confirmation) =>
+            new MultisigConfirmationDetails(
+              new AddressInfo(confirmation.owner),
+              confirmation.signature,
+              confirmation.submissionDate.getTime(),
+            ),
+        );
+    const refundReceiver = await this.addressInfoHelper.getOrDefault(
+      chainId,
+      transaction.refundReceiver ?? NULL_ADDRESS,
+    );
+    const rejectors =
+      txStatus === TransactionStatus.Cancelled
+        ? await this._getRejectors(chainId, transaction, safe)
+        : null;
+
+    return {
+      submittedAt: transaction.submissionDate.getTime(),
+      nonce: transaction.nonce,
+      safeTxGas: transaction.safeTxGas?.toString() ?? '0',
+      baseGas: transaction.baseGas?.toString() ?? '0',
+      gasPrice: transaction.gasPrice?.toString() ?? '0',
+      refundReceiver,
+      safeTxHash: transaction.safeTxHash,
+      executor,
+      signers,
+      confirmationsRequired,
+      confirmations,
+      rejectors,
+      gasToken,
+      gasTokenInfo,
+      trusted: transaction.trusted,
+    };
+  }
+
+  private async _getRecipientAddressInfo(
```

Comment:
The recipient could be a token contract or a contract. Trying to get the data with `Token` as source and fallback to `Contract` as source.